### DISCUSSION TITLE
Coarse-graining of arbitrary protein sequences

### DISCUSSION THREAD

**Discussion by freakyacorn** [marrink-lab/vermouth-martinize#605] [Q&A]
I would like to use martinize2 to coarse-grain arbitrary protein sequences. I will be operating under the assumption that these sequences have no secondary structure to worry about. Right now, I am starting with an all-atom .pdb file constructed by simply bonding together amino acids and adding head/tail groups. I can ensure there are no interlocking rings in the resulting structure. The questions I have are  meant to ensure that martinize2 is generating the correct coarse-grained mapping.

1. If the -bonds-from flag is set to "none", martinize2 says it will only use bonds from the "CONECT" portion of the input file. Assuming that the correct topology has been outlined in this section of the .pdb, will martinize2 generate the correct mapping and topology? Will it generate the correct mapping even if the distances in the file are unphysical?
2. If the -bonds-from flag is set to "distance", how does one ensure that the distances in the .pdb file are sensible, particularly for an arbitrary sequence? Is there a particular force field that is most sensible? How about protein conformation?
3. In what way do the atoms in the .pdb file of a protein need to be named to ensure that the correct mapping is applied for each residue?

Any help with these questions is greatly appreciated.

**Comment by pckroon** [comment]
Hi Wesley,

I'll (try to) answer your questions point by point.
1. Yes, but some parameters might depend on the (unphysical) coordinates. The prime example of this are elastic bonds, another are the scfix parameters (https://github.com/marrink-lab/vermouth-martinize/blob/master/vermouth/data/force_fields/martini3001/aminoacids.ff#L899). At a quick glance I can't spot any other, but no promises ;) The easiest check would be to generate 2 topologies from 2 random initial coordinates, and see if they're different. (@fgrunewald Should we print a message (of some sort) whenever we generate parameters from coordinates?)
2. A minimization should be sufficient, for example using UFF. You don't need anything fancy. We guess bonds using the exact same algorithm VMD uses, so you can use that (or `-write-graph`) to check.
3. Very little :) We identify atoms primarily based on connectivity. You need to make sure your residue names are correct, and the combination `(chain, resid, rename, insertion_code, molecule)` is unique for all residues. You also need to check that the element of all atoms is correct (or at least, the first letter of the atomname matches the element; beware chlorine and iron). Beyond that atomnames are optional. We overlay residues with their reference such that most atomnames match. You may have issues with serine, since without atomnames there is a wrong solution where O gets identified as OG. As a sidenote, the more atomnames are correct, the faster Martinize2 will run.

Finally, have a look at https://github.com/marrink-lab/polyply_1.0. Polyply will not only generate the topology for you, but can also generate decent coordinates from scratch.

**Reply by freakyacorn** [reply]
Hi Peter, thanks for the detailed reply. I have two questions/issues regarding using polyply to generate .itp files and starting configurations.

1. One issue I am running into is in the generation of starting coordinates from the .itp file created with the gen_params command. When doing this for a protein sequence, each bead has an associated virtual site (VS). When I try and run the gen_coords command, the VSs are not recognized. I can include the VS atomtype and nonbonded parameter with water in the forcefield definition to fix this, but I'd rather just not have them at all. Is there a way to disable VS generation?
2. Let's say I generate a .itp file using gen_params, create coordinates for the resulting coarse-grained structure using gen_coords, and then perform a decent backmapping to the all-atom structure. If I then minimize and coarse-grain that all-atom structure, will the re-coarse-grained molecule/.itp be the same as what was created by gen_params? In other words, are coarse-grained representations of residues sequence independent (excluding head and tail groups)? Is this the same for other polymer materials?

Thank you again for your help.

**Reply by pckroon** [reply]
1. For this I really have to direct you to the polyply discussion forum. In general though, your virtual sites *are* an integral part of your model, and without them your simulations will become unphysical.
2. Complex question. It should. The itp and coordinates that polyply generates for you are (of course) strongly related in that they are from the same model. Martinize2 is capable of mapping AA structures to CG models; assuming both implementations are correct, and your backmapping is good enough, you should end up with the same itp and CG structure as you'd get from polyply directly. I'm not sure what you mean by "are coarse-grained representations of residues sequence independent"; there are sequence dependent parameters, for example the angle potential across a proline (X-P-Y; didn't check, but makes sense to me). The *mapping* is sequence independent, and as far as I know this is true for all martini polymer models.

**Reply by fgrunewald** [reply]
@freakyacorn did you install the latest version from GitHub? @csbrasnett is the expert who might be able to figure this one out

**Reply by csbrasnett** [reply]
@freakyacorn feel free to drop me a mail with any questions about the model. The polyply wiki has a small [tutorial](https://github.com/marrink-lab/polyply_1.0/wiki/Tutorial:-Martini-3-IDPs) about constructing proteins in polyply, and the associated [preprint](https://doi.org/10.1101/2024.04.15.589479) may also be useful for context.
